plugins/power/tasmota_

```diff
@@ -0,0 +1,155 @@
+# ln -s /usr/share/munin/plugins/tasmota_ /etc/munin/plugins/tasmota_hostname_current
+#
+# Plugin also implements suggests, so if you have /etc/genders
+# populated with "tasmota,powermon" flags for those tasmota devices
+# that implement energy monitoring, and nodeattr installed, you can
+# run an ansible play like this to set up your links:
+# https://github.com/spacelama/ansible-initial-server-setup/tree/master/roles/monitoring/tasks
+#
+#%# family=auto
+#%# capabilities=autoconf
+
+SWITCH=$(basename "$0" | cut -d_ -f2)
+FUNCTION=$(basename "$0" | cut -d_ -f3)
+
+if [ "$1" = "autoconf" ]; then
+    echo yes && exit 0
+fi
+
+if [ "$1" = "suggest" ]; then
+    nodeattr -n 'tasmota&&powermon' | while read device ; do
+	for i in voltage power powerfactor current energy; do
+	    echo "${device}_${i}"
+	done
+    done
+    exit
+fi
+
+voltage() {
+    if [ "$1" = "config" ]; then
+
+        echo "graph_title Tasmota Voltage: $SWITCH"
+        echo "graph_category sensors"
+        echo "graph_vlabel Volts"
+        echo "graph_args --base 1000 -l 0"
+
+	echo "Volts.label Volts"
+        exit 0
+    else
+        res=$( curl -fsS --data-urlencode "cmnd=Status 8" http://$SWITCH/cm )
+        volts=$( echo "$res" | jq -c '.StatusSNS.ENERGY.Voltage' )
+        echo Volts.value $volts
+    fi
+}
+
+current() {
+    if [ "$1" = "config" ]; then
+
+        echo "graph_title Tasmota Current: $SWITCH"
+        echo "graph_category sensors"
+        echo "graph_vlabel Amps"
+        echo "graph_args --base 1000 -l 0"
+
+	echo "Current.label Current"
+        exit 0
+    else
+        res=$( curl -fsS --data-urlencode "cmnd=Status 8" http://$SWITCH/cm )
+        current=$( echo "$res" | jq -c '.StatusSNS.ENERGY.Current' )
+        echo Current.value $current
+    fi
+}
+
+power() {
+    if [ "$1" = "config" ]; then
+
+	echo "graph_title Tasmota Power: $SWITCH"
+        echo "graph_category sensors"
+	echo "graph_vlabel Watts"
+        echo "graph_args --base 1000 -l 0"
+
+	for i in Power ApparentPower ReactivePower ; do
+	    echo "${i}.label $i"
+	    echo "${i}.type GAUGE"
+	    echo "${i}.min 0"
+	done
+    else
+        res=$( curl -fsS --data-urlencode "cmnd=Status 8" http://$SWITCH/cm )
+	for i in Power ApparentPower ReactivePower ; do
+            power=$( echo "$res" | jq -c '.StatusSNS.ENERGY.'$i )
+            echo $i.value $power
+        done
+    fi
+}
+
+powerfactor() {
+    if [ "$1" = "config" ]; then
+
+        echo "graph_title Tasmota Power factor: $SWITCH"
+        echo "graph_category sensors"
+        echo "graph_vlabel Power Factor"
+        echo "graph_args --base 1000 -l 0"
+
+	echo "PowerFactor.min 0"
+	echo "PowerFactor.max 1"
+	echo "PowerFactor.label Power Factor"
+        exit 0
+    else
+        res=$( curl -fsS --data-urlencode "cmnd=Status 8" http://$SWITCH/cm )
+        powerfactor=$( echo "$res" | sed 's/.*"Factor":\([0-9.]*\)[^0-9.]*.*/\1/' )
```

Comment:
Maybe you can parse this with `jq`, too?
If not: please add a comment with an example of the expected format.

Reply:
Done.